diesel/src/pg/types/numeric.rs

```diff
@@ -0,0 +1,111 @@
+#[cfg(feature="bigdecimal")]
+mod bigdecimal {
+    extern crate num_traits;
+    extern crate num_bigint;
+    extern crate num_integer;
+    extern crate bigdecimal;
+
+    use std::error::Error;
+    use std::io::prelude::*;
+
+    use pg::Pg;
+
+    use self::num_traits::{Signed, Zero, ToPrimitive};
+    use self::num_bigint::{Sign, BigInt, BigUint};
+    use self::num_integer::Integer;
+    use self::bigdecimal::BigDecimal;
+
+    use pg::data_types::PgNumeric;
+    use types::{self, FromSql, ToSql, IsNull};
+
+    impl ToSql<types::Numeric, Pg> for BigDecimal {
+        fn to_sql<W: Write>(&self, out: &mut W) -> Result<IsNull, Box<Error + Send + Sync>> {
+            use std::str::FromStr;
+
+            let absolute = self.abs();
+
+            let mut digits = vec![];
+            let ten_k = BigInt::from(10000);
+            let mut integer_part = absolute.to_bigint();
+            let mut decimal_part = absolute - &integer_part;
+            // scale is the amount of digits to print. to_string() includes a "0.",
+            // that's why the -2 is there.
+            let scale = decimal_part.to_string().len() as u16 - 2;
+            let mut weight = 0;
+
+            // Encode the integer part
+            while ten_k < integer_part {
+                weight += 1;
+                // digit is integer_part REM 10_000
+                let (div, digit) = integer_part.div_rem(&ten_k);
+                digits.push(digit.to_i16().unwrap());
```

Comment:
Please the `unwrap`s with `expect`s that include a reason why this is a failure that should be allowed to crash the process, or propagate the error (feel free to crate a new error type if necessary) :)